pkg/middleware/auth.go

```diff
@@ -99,6 +102,95 @@ func (c JWTVerifier) CheckToken(token string) (*jwt.Token, error) {
 	return nil, lastSeenErr
 }
 
+// JWTVerifierURL checks token with JWT against JWKS, implements Verifier
+type JWTVerifierURL struct {
+	jwksURL string
+
+	parser   *jwt.Parser
+	jwksKeys *jwk.Set
+	jwksLock sync.RWMutex
+}
+
+func (c *JWTVerifierURL) startTicker(ticker time.Duration) {
+
+	for range time.Tick(ticker) {
+		err := c.updateKeySet()
+		if err != nil {
+			log.Warn().Msg("unable to update JWKS key set")
```

Comment:
Should add the error to the log message:
```
log.Warn().Err(err).Msg("unable to update JWKS key set")
```

Reply:
I think this one should stay as a warning, because we don't terminate the program

Reply:
this is still a warning (e.g. `.Warn()`), we're just adding the error to the output. Note `.Err(err)` is **not** a log level but a way to append the error message. Basically shorthand for `.Str("error", err.Error())`.

Reply:
ok, I missed that